server/src/main/java/org/apache/druid/server/QueryLifecycle.java

```diff
@@ -220,6 +220,18 @@ public void initialize(final Query<?> baseQuery)
    * @return authorization result
    */
   public Access authorize(HttpServletRequest req)
+  {
+    return authorize(AuthorizationUtils.authenticationResultFromRequest(req));
+  }
+
+  /**
+   * Authorize the query using the authentication result.
+   * Will return an Access object denoting whether the query is authorized or not.
+   *
+   * @param authenticationResult authentication result indicating identity of the requester
+   * @return authorization result of requester
+   */
+  public Access authorize(AuthenticationResult authenticationResult)
```

Comment:
Please include some unit tests for this new method, in case it's called directly.

Reply:
I have updated the existing tests for `authorize(HttpServletRequest)` to also test this method.